apps/server/src/modules/account/services/account-idm.service.integration.spec.ts

```diff
@@ -93,79 +91,125 @@ describe('AccountIdmService Integration', () => {
 		}
 	});
 
-	it('save should create a new account', async () => {
-		if (!isIdmReachable) return;
-		const createdAccount = await accountIdmService.save(testAccount);
-		const foundAccount = await identityManagementService.findAccountById(createdAccount.idmReferenceId ?? '');
-
-		expect(foundAccount).toEqual(
-			expect.objectContaining<IdmAccount>({
-				id: createdAccount.idmReferenceId ?? '',
-				username: createdAccount.username,
-				attDbcAccountId: testDbcAccountId,
-				attDbcUserId: createdAccount.userId,
-				attDbcSystemId: createdAccount.systemId,
-			})
-		);
+	describe('save', () => {
+		describe('when account does not exists', () => {
+			it('should create a new account', async () => {
+				if (!isIdmReachable) return;
+				const createdAccount = await accountIdmService.save(testAccount);
+				const foundAccount = await identityManagementService.findAccountById(createdAccount.idmReferenceId ?? '');
+
+				expect(foundAccount).toEqual(
+					expect.objectContaining<IdmAccount>({
+						id: createdAccount.idmReferenceId ?? '',
+						username: createdAccount.username,
+						attDbcAccountId: createdAccount.id,
+						attDbcUserId: createdAccount.userId,
+						attDbcSystemId: createdAccount.systemId,
+					})
+				);
+			});
+		});
 	});
 
-	it('save should update existing account', async () => {
-		if (!isIdmReachable) return;
-		const newUsername = 'jane.doe@mail.tld';
-		const idmId = await createAccount();
-
-		await accountIdmService.save({
-			id: testDbcAccountId,
-			username: newUsername,
+	describe('save', () => {
+		describe('when account exists', () => {
+			const setup = async () => {
+				const newUserName = 'jane.doe@mail.tld';
+				const idmId = await createAccount();
+
+				return { idmId, newUserName };
+			};
+			it('should update account', async () => {
+				if (!isIdmReachable) return;
```

Comment:
What has account to do with the idm in the end? Behind this we need a mock that work like the idm.